### PR TITLE
SAAS-820 add UI test to create new ticket in sn

### PR DESCRIPTION
**Pull request by peterSirotnak**
For ticket SAAS-820: 

https://jira.percona.com/browse/SAAS-820

### CODE REVIEW

File: cypress/integration/freeUser/dashboard/dashboard.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  });
+
+  it('SAAS-T198 - Verify free account user is not able to get organization tickets', () => {
+    cy.contains(gettingStartedPage.constants.labels.addOrganization, { timeout: timeouts.HALF_MIN })
+      .should('be.visible')
+      .click();
+    let orgName = faker.company.companyName();
+    cy.findByTestId(organizationPage.locators.organizationNameInput).type(orgName);
+    cy.findByTestId(organizationPage.locators.createOrganizationButton).click();
+    verifyOrganizationName(orgName);
```
Comment (yurkovychv) [inline cypress/integration/freeUser/dashboard/dashboard.spec.js:25]: The test code on `19-25` lines can be replaced with `cy.apiCreateOrg` and moved to `beforeEach` hook. We should use `API` calls as much as possible unless we want to really test that functionality

File: cypress/integration/customer/dashboard/dashboard.spec.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      cy.findByTestId(dashboardPage.locators.ticketSection)
+        .find('a')
+        .hasAttr('target', '_blank')
+        .hasAttr('href', setServiceNowAddress(browserURL));
+    });
+    cy.logoutUser();
+    cy.oktaCreateUser(technicalUser);
+    cy.loginByOktaApi(technicalUser.email, technicalUser.password);
+    openViewOrganizationPage();
+    cy.findAllByTestId(commonPage.locators.sideMenuLink).get('a[href*="/dashboard"]').click();
```
Comment (yurkovychv) [inline cypress/integration/customer/dashboard/dashboard.spec.js:37]: Also, you can use `[].forEach` to have separate tests for `admin` and `technical` roles

File: cypress/integration/customer/dashboard/dashboard.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { commonPage } from 'pages/common.page';
+import dashboardPage from 'pages/dashboard.page';
+import { openViewOrganizationPage } from '../organizations/helper';
+import { setServiceNowAddress } from './helper';
+
+context('Dashboard Tests for customers', () => {
+  let admin1User;
+  let admin2User;
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js]: Removed

File: cypress/integration/customer/dashboard/dashboard.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  let technicalUser;
+  let organizationName;
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js]: Removed

File: cypress/integration/customer/dashboard/helper.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import dashboardPage from 'pages/dashboard.page';
+
+export const setServiceNowAddress = (location) => {
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/helper.js]: production url removed

File: cypress/integration/customer/dashboard/dashboard.spec.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      admin2User = account.admin1;
+      technicalUser = account.technical;
+      organizationName = account.name;
+    });
+  });
+
+  it('SAAS-T233 - Verify "open new ticket" link for Percona customer', () => {
+    cy.oktaCreateUser(admin1User);
+    cy.loginByOktaApi(admin1User.email, admin1User.password);
+    openViewOrganizationPage();
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js]: Changed to api call and added to beforeEach hooks.

File: cypress/integration/freeUser/dashboard/dashboard.spec.js (hunk 1 above)
Comment (peterSirotnak) [reply cypress/integration/freeUser/dashboard/dashboard.spec.js]: Changed to api call and added to beforeEach hooks.

File: cypress/integration/customer/dashboard/dashboard.spec.js (hunk 2 above)
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js]: Good idea! changed to forEach loop.

File: cypress/support/api/okta_commands.js [hunk 7]
```diff
@@ -46,6 +46,10 @@ Cypress.Commands.add('loginByOktaApi', (username, password) =>
     }),
 );
 
+Cypress.Commands.add('retrieveUserAccessToken', () => {
```
Comment (yurkovychv) [inline cypress/support/api/okta_commands.js:49]: why not to use `cy.getUserAccessToken`? 
If we want to keep it for simplicity may be worth it to be called `retrieveCurrentUserAccessToken`

File: cypress/integration/customer/dashboard/dashboard.spec.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+  it('SAAS-T233 - Verify "open new ticket" link for Percona customer', async () => {
+    users.forEach((user) => {
+      cy.loginByOktaApi(user.email, user.password);
+      cy.findAllByTestId(commonPage.locators.sideMenuLink).get('a[href*="/dashboard"]').click();
+      cy.findByTestId(dashboardPage.locators.ticketSection)
+        .find('a')
+        .hasAttr('target', '_blank')
+        .hasAttr('href', dashboardPage.links.serviceNowAddress);
+      cy.logoutUser();
```
Comment (yurkovychv) [inline cypress/integration/customer/dashboard/dashboard.spec.js:25]: I think logout is not necessary here. Could you please double-check without it?

Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js:25]: Works without it. But without it we are just replacing token in the local storage not removing the old one.

File: cypress/support/api/okta_commands.js (hunk 7 above)
Comment (peterSirotnak) [reply cypress/support/api/okta_commands.js:49]: `cy.getUserAccessToken` adds another api call for data that we already have present.  `retrieveCurrentUserAccessToken` retrieves token from local storage

File: cypress/integration/customer/dashboard/dashboard.spec.js (hunk 8 above)
Comment (yurkovychv) [reply cypress/integration/customer/dashboard/dashboard.spec.js:25]: Isn't it the same ? I mean if we will Log out and log in with another user the local storage item will be deleted and created again. For me, it's the same as we will just over write it

File: cypress/support/api/okta_commands.js (hunk 7 above)
Comment (peterSirotnak) [reply cypress/support/api/okta_commands.js]: Renamed command.

File: cypress/integration/customer/dashboard/dashboard.spec.js (hunk 8 above)
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js]: Okay, created method to remove token from local storage to make it quick.

**Review by yurkovychv — APPROVED** [review]
Great job @peterSirotnak ! Just one minor comment

File: cypress/support/ui/common.js [hunk 9]
```diff
@@ -59,3 +60,10 @@ Cypress.Commands.add('getTable', { prevSubject: true }, (subj, options = {}) =>
     }, {}),
   );
 });
+
+// Logs user out of the platform
```
Comment (atymchuk) [inline cypress/support/ui/common.js:64]:
```suggestion
// Logs user out of the Portal
```

File: cypress/support/ui/common.d.ts [hunk 10]
```diff
@@ -65,5 +65,13 @@ declare namespace Cypress {
      *   });
      */
     getTable(options: {}): Chainable;
+
+    /**
+     * Logs user out of the Percona Platform.
```
Comment (atymchuk) [inline cypress/support/ui/common.d.ts:70]:
```suggestion
     * Logs user out of the Portal.
```

File: cypress/integration/customer/dashboard/dashboard.spec.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+      users.push(account.admin1, account.technical);
+      users.forEach((user) => cy.oktaCreateUser(user));
+      createOrgAndAddUsers(users);
+    });
+  });
+
+  it('SAAS-T233 - Verify "open new ticket" link for Percona customer', () => {
+    users.forEach((user) => {
+      if (user === users[0]) cy.log('Running test for Admin user');
+      if (user === users[1]) cy.log('Running test for Technical user');
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply cypress/integration/customer/dashboard/dashboard.spec.js]: Okay, thank you and updated log message.

File: cypress/support/ui/common.js (hunk 9 above)
Comment (peterSirotnak) [reply cypress/support/ui/common.js]: Fixed.

File: cypress/support/ui/common.d.ts (hunk 10 above)
Comment (peterSirotnak) [reply cypress/support/ui/common.d.ts]: Fixed.

**Review by atymchuk — APPROVED** [review]

**Comment by atymchuk** [comment]
Great job, Peter!

**Pull request merged**